gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/Text.java

```diff
@@ -29,6 +31,48 @@
  */
 public enum Text implements BiPredicate<String, String> {
 
+    /**
+     * Evaluates if the first string has a regex match with the second (pattern).
+     *
+     * @since 3.6.0
+     */
+    regex {
+        @Override
+        public boolean test(final String value, final String regex) {
+            Pattern pattern = Pattern.compile(regex);
```

Comment:
Using

Pattern.compile followed by Matcher matcher = pattern.matcher(value); is equivalent to using `String.matches(regex)`. We might was well use the later since it simplifies code.

The Pattern + Matcher approach is better if we plan to re-use the compiled pattern.

Reply:
The reason I left it this way is related to your other comment. As the code paths I believe are going to be almost identical, this sets us up to potentially cache Pattern objects in the future.